x-pack/legacy/plugins/apm/common/projections/metrics.ts

```diff
@@ -19,24 +24,36 @@ export function getMetricsProjection({
 }) {
   const { start, end, uiFiltersES, config } = setup;
 
-  const serviceNodeNameFilters = serviceNodeName
-    ? [{ term: { [SERVICE_NAME]: serviceName } }]
-    : [];
+  const filter = [
+    { term: { [SERVICE_NAME]: serviceName } },
+    { term: { [PROCESSOR_EVENT]: 'metric' } },
+    { range: rangeFilter(start, end) },
+    ...uiFiltersES
+  ];
+
+  let mustNot = {};
+
+  if (serviceNodeName && serviceNodeName === SERVICE_NODE_NAME_MISSING) {
+    mustNot = {
+      must_not: [
+        {
+          exists: { field: SERVICE_NODE_NAME }
+        }
+      ]
+    };
+  } else if (serviceNodeName) {
+    filter.push({
+      term: { [SERVICE_NODE_NAME]: serviceNodeName }
+    });
+  }
```

Comment:
You can keep it all in a single filter - but it comes at the cost of a nested ternary.

```js
const serviceNodeNameFilters = serviceNodeName
  ? serviceNodeName === SERVICE_NODE_NAME_MISSING
    ? { bool: { must_not: [{ exists: { field: SERVICE_NODE_NAME } }] } }
    : { term: { [SERVICE_NODE_NAME]: serviceNodeName } }
  : [];
```

Reply:
Alternatively:
```ts
function getServiceNodeNameFilters(serviceNodeName?: string) {
  if (!serviceNodeName) {
    return [];
  }

  return serviceNodeName === SERVICE_NODE_NAME_MISSING
    ? { bool: { must_not: [{ exists: { field: SERVICE_NODE_NAME } }] } }
    : { term: { [SERVICE_NODE_NAME]: serviceNodeName } };
}
```

Reply:
Nice, I did not consider using a nested `bool`. I'll take the latter.